### PR TITLE
Add more comments for clang-format

### PR DESCRIPTION
**Pull request by zhangbo1882**
Add more comments for clang-format

Signed-off-by: bozhang <bozhang@ebay.com>

Refe to https://github.com/envoyproxy/envoy/pull/20175

### CODE REVIEW

File: bazel/README.md [hunk 1]
```diff
@@ -916,7 +916,19 @@ Edit the paths shown here to reflect the installation locations on your system:
...
+The best way to get the proper clang-format is to copy the clang-format from the ci docker image.
+* Run the ci docker image
+```shell
+ci/run_envoy_docker.sh bash
+```
+* Get the docker container ID and copy the clang-format to host machine
+```shell
+docker  copy $dockerContainerID:/opt/llvm/bin/clang-format clang-format-ci
+```
+* Replace the host clang-format with the new one. You can create a symbolic link to point to the clang-format-ci.
```
Comment (phlax) [inline bazel/README.md:928]: Im wondering if a bit more advice would be helpful - something like...

> Ensure that the copied `clang-format` is the default one. You can do this by ensuring it is in `$PATH` or by creatig a symbolic link:

**Review by phlax — CHANGES_REQUESTED** [review]
Thanks for this @zhangbo1882 - a few nits left inline

File: bazel/README.md [hunk 2]
```diff
@@ -916,7 +916,19 @@ Edit the paths shown here to reflect the installation locations on your system:
 export CLANG_FORMAT="$HOME/ext/clang+llvm-12.0.1-x86_64-linux-gnu-ubuntu-16.04/bin/clang-format"
 export BUILDIFIER_BIN="/usr/bin/buildifier"
 ```
-
+The best way to get the proper clang-format is to copy the clang-format from the ci docker image.
```
[the remark this answers is not in the document]
Comment (zhangbo1882) [reply bazel/README.md:919]: will fix

File: bazel/README.md [hunk 3]
```diff
@@ -916,7 +916,19 @@ Edit the paths shown here to reflect the installation locations on your system:
...
+* Run the ci docker image
+```shell
+ci/run_envoy_docker.sh bash
+```
+* Get the docker container ID and copy the clang-format to host machine
```
[the remark this answers is not in the document]
Comment (zhangbo1882) [reply bazel/README.md:924]: will add

File: bazel/README.md (hunk 1 above)
Comment (zhangbo1882) [reply bazel/README.md]: will fix

File: bazel/README.md [hunk 4]
```diff
@@ -916,16 +916,20 @@ Edit the paths shown here to reflect the installation locations on your system:
...
+* Get the docker container ID
+```shell
+docker ps | grep envoy-build-ubuntu | awk '{print $1}'
+```
+* Copy the `clang-format` to host machine
 ```shell
 docker  copy $dockerContainerID:/opt/llvm/bin/clang-format clang-format-ci
 ```
-* Replace the host clang-format with the new one. You can create a symbolic link to point to the clang-format-ci.
+* Replace the host `clang-format` with the new one. Ensure that the copied `clang-format` is the default one. You can do this by ensuring it is in $PATH or by creatig a symbolic link:
```
Comment (phlax) [inline bazel/README.md:932]: sorry my typo - `creatig` -> `creating`

also probs surround `$PATH` with backticks

File: bazel/README.md [hunk 5]
```diff
@@ -916,7 +916,27 @@ Edit the paths shown here to reflect the installation locations on your system:
...
 ```
-
+The easiest way to use the correct `clang-format` in your host system is to copy the `clang-format` from the ci docker image.
+* Run the ci docker image
+```shell
+ci/run_envoy_docker.sh bash
+```
+* Get the docker container ID
+```shell
+docker ps | grep envoy-build-ubuntu | awk '{print $1}'
```
Comment (phlax) [inline bazel/README.md:926]: v minor nit (thanks for iterating so far!)

maybe set the `dockerContainerID` here like...

```
dockerContainerID=$(docker ps | grep envoy-build-ubuntu | awk '{print $1}')
```

it just means people can copypasta their way through

File: bazel/README.md [hunk 6]
```diff
@@ -916,7 +916,27 @@ Edit the paths shown here to reflect the installation locations on your system:
...
+```shell
+dockerContainerID=$(docker ps | grep envoy-build-ubuntu | awk '{print $1}')
+```
+* Copy the `clang-format` to host machine
+```shell
+docker  copy $dockerContainerID:/opt/llvm/bin/clang-format clang-format-ci
+```
+* Replace the host `clang-format` with the new one. Ensure that the copied `clang-format` is the default one. You can do this by ensuring it is in `$PATH` or by creating a symbolic link:
+```shell
+copy clang-format-ci $PATH/clang-format
```
Comment (phlax) [inline bazel/README.md:934]: `$PATH` cant be used as a path as it has multiple paths

i think the cannonical way to do this is probably to copy the bin to `/usr/local/bin` - ie

```console
cp clang-format-ci /usr/local/bin/clang-format
```

or as a (non-root) alternative using a bin dir and adding that to `$PATH`

```
mkdir bin
mv clang-format-ci bin/
PATH=$PATH:$PWD/bin/
```

File: bazel/README.md [hunk 7]
```diff
@@ -931,12 +931,18 @@ docker  copy $dockerContainerID:/opt/llvm/bin/clang-format clang-format-ci
...
 * Replace the host `clang-format` with the new one. Ensure that the copied `clang-format` is the default one. You can do this by ensuring it is in `$PATH` or by creating a symbolic link:
 ```shell
-copy clang-format-ci $PATH/clang-format
+cp clang-format-ci /usr/local/bin/clang-format
 ```
 or
 ```
 ln -s clang-format-ci $PATH/clang-format
 ```
+If you are a non-root user, alternative you can use a bin dir and add that to `$PATH`
```
Comment (phlax) [inline bazel/README.md:940]: s/alternative/alternatively/

File: bazel/README.md [hunk 8]
```diff
@@ -931,12 +931,18 @@ docker  copy $dockerContainerID:/opt/llvm/bin/clang-format clang-format-ci
...
+```shell
+mkdir bin
+mv clang-format-ci bin/
```
Comment (phlax) [inline bazel/README.md:943]: this needs to be `mv clang-format-ci bin/clang-format` to be available by correct name

File: bazel/README.md [hunk 9]
```diff
@@ -931,12 +931,18 @@ docker  copy $dockerContainerID:/opt/llvm/bin/clang-format clang-format-ci
...
+PATH=$PATH:$PWD/bin/
```
Comment (phlax) [inline bazel/README.md:944]: i think we probably need to export PATH in this case so

```
export  PATH=$PATH:$PWD/bin/
```

File: bazel/README.md [hunk 10]
```diff
@@ -916,7 +916,33 @@ Edit the paths shown here to reflect the installation locations on your system:
...
+```shell
+docker  copy $dockerContainerID:/opt/llvm/bin/clang-format clang-format-ci
+```
+* Replace the host `clang-format` with the new one. Ensure that the copied `clang-format` is the default one. You can do this by ensuring it is in `$PATH` or by creating a symbolic link:
+```shell
+cp clang-format-ci /usr/local/bin/clang-format
+```
+or
+```
+ln -s clang-format-ci $PATH/clang-format
```
Comment (phlax) [inline bazel/README.md:938]: i think just remove this bit - it wont work as it stands

Comment (zhangbo1882) [reply bazel/README.md:938]: I can remove it. But why it can not work?

Comment (phlax) [reply bazel/README.md]: try this

```console
$ echo $PATH/clang-format
```

**Review by phlax — APPROVED** [review]
thanks @zhangbo1882 for working through this

**Pull request merged**